seed/serialize.py

```diff
@@ -44,6 +112,57 @@ def validate(seed):
             print("platform not in ", PLATFORMS)
             return False
 
+    feature_names_to_studies = collections.defaultdict(list)
+    for study in seed['studies']:
+        used_feature_names = set()
+        for experiment in study['experiments']:
+            feature_association = experiment.get('feature_association')
+            if feature_association:
+                for enable_feature in feature_association.get('enable_feature', []):
+                    used_feature_names.add(enable_feature)
+                for disable_feature in feature_association.get('disable_feature', []):
+                    used_feature_names.add(disable_feature)
+
+        for used_feature_names in used_feature_names:
+            feature_names_to_studies[used_feature_names].append(study)
+
+    def get_study_platforms(study):
+        return set(study.get('filter', {}).get('platform', []))
+
+    def get_study_channels(study):
+        return set(study.get('filter', {}).get('channel', []))
+
+    def get_study_version_range(study):
+        return [
+            version_to_int_array(study.get('filter', {}).get('min_version')),
+            version_to_int_array(study.get('filter', {}).get('max_version')),
+        ]
+
+    def is_set_intersect(a, b):
+        return a == b or a.intersection(b)
+
+    def is_version_range_intersect(range1, range2):
+        return compare_versions(range1[1], range2[0]) >= 0 and compare_versions(range2[1], range1[0]) >= 0
+
+    test_version_comparison()
+    for studies in feature_names_to_studies.values():
+        for i, study1 in enumerate(studies):
+            study1_platform = get_study_platforms(study1)
```

Comment:
Could `study_platform` or `study_channel` be empty?

Reply:
> Could `study_platform` or `study_channel` be empty?

yep. improved the intersection check to cover this.